ovs/dal/datalist.py

```diff
@@ -103,16 +116,132 @@ def guids(self):
             self._execute_query()
         return self._guids
 
+    def set_key(self, key=None):
+        """
+        Sets the caching key
+        Won't override the key when a key was giving on initializing
+        :param key: Key to explicitly use
+        :type key: str
+        :return: None
+        :rtype: NoneType
+        """
+        if key is not None:
+            self._key = '{0}_{1}'.format(DataList.NAMESPACE, key)
+            self._provided_key = True
+            # Unsure whether or not the same query would apply
+            self._volatile.delete(self._key)
+            return
+        elif self._provided_key is False:
```

Comment:
As far as i can tell, its impossible to reset the key
If you create a DataList without a key --> `self._provided_key` will be `False`
If you then use `set_key` on that DataList --> `self._provided_key` will be `True`
But there is no way to reset the provided key right?  Not without tampering with the private properties